### ISSUE TITLE
Compile error in core functions

### ISSUE THREAD

**Issue by incrediblesound** [incrediblesound/huo#54]
``` shell
src/core_functions.c:243:90: error: format specifies type 'uintmax_t' (aka 'unsigned long') but the argument has type 'unsigned long long' [-Werror,-Wformat]
        ERROR("Index out of range for set: should be 0 <= %" PRIhi " < %"PRIuMAX, index, SIZE_MAX);
                                                                                         ^~~~~~~~
/usr/include/stdint.h:153:20: note: expanded from macro 'SIZE_MAX'
#define SIZE_MAX          UINT64_MAX
                          ^~~~~~~~~~
/usr/include/stdint.h:87:27: note: expanded from macro 'UINT64_MAX'
#define UINT64_MAX        18446744073709551615ULL
                          ^~~~~~~~~~~~~~~~~~~~~~~
src/base_util.h:41:59: note: expanded from macro 'ERROR'
#define ERROR(...) ERROR_AT(__FILE__, __func__, __LINE__, __VA_ARGS__)
                                                          ^
src/base_util.h:51:21: note: expanded from macro 'ERROR_AT'
    fprintf(stderr, __VA_ARGS__);\
```

**Comment by incrediblesound** [comment]
@TheLoneWolfling  could you take a look at this one?

**Comment by TheLoneWolfling** [comment]
Whoops.

PRIuMAX should be %zu instead.

I'll fix it.

**Comment by TheLoneWolfling** [comment]
Closed by commit 8dd88ce
